### PR TITLE
Bug 1867717 - Add DownloadsFeature to AddonPopupBaseFragment

### PR DESCRIPTION
**Pull request by Amejia481**
### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1867717

### CODE REVIEW

File: android-components/components/feature/webcompat/src/main/java/mozilla/components/feature/webcompat/WebCompatFeature.kt [hunk 1]
```diff
@@ -14,7 +14,7 @@ object WebCompatFeature {
     private val logger = Logger("mozac-webcompat")
 
     internal const val WEBCOMPAT_EXTENSION_ID = "webcompat@mozilla.org"
-    internal const val WEBCOMPAT_EXTENSION_URL = "resource://android/assets/extensions/webcompat/"
+    internal const val WEBCOMPAT_EXTENSION_URL = "resource://android/assets/extensions/extension/"
```
Comment (Amejia481) [inline android-components/components/feature/webcompat/src/main/java/mozilla/components/feature/webcompat/WebCompatFeature.kt:17]: This was just for testing

**Comment by Amejia481** [comment]
This is ready for review :)

**Comment by willdurand** [comment]
I don't think I can review this patch, this is a bit out of my league. @jonalmeida is that something you could look at please?

**Comment by Amejia481** [comment]
@jonalmeida would you mind taking a look at this ? :)

File: fenix/app/src/test/java/org/mozilla/fenix/browser/DownloadDialogUtilsTest.kt [hunk 2]
```diff
@@ -0,0 +1,297 @@
...
+import org.junit.Assert.assertTrue
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@RunWith(AndroidJUnit4::class)
+class DownloadDialogUtilsTest {
+
+    @Test
+    fun `GIVEN final status and openInApp WHEN calling handleOnDownloadFinished and try to open the file fails THEN show an error message`() {
+        val downloadUtils = spyk(DownloadDialogUtils)
```
Comment (jonalmeida) [inline fenix/app/src/test/java/org/mozilla/fenix/browser/DownloadDialogUtilsTest.kt:29]: Can we avoid using a spy? It is very easy to misuse them and if we are relying on them then I believe we need to spend more time thinking about how we structure our code to be testable. For example, finding a way to delegate out the `FenixSnackbar` code in a test.

In fact, on lines 47 to 53, we should not be using the spy and are no longer executing the real code path needed to catch real regressions.

```
        every {
            downloadUtils.shouldShowCompletedDownloadDialog(
                any(),
                any(),
                any(),
            )
        } returns true
```

File: fenix/app/src/main/java/org/mozilla/fenix/addons/AddonPopupBaseFragment.kt [hunk 3]
```diff
@@ -51,6 +63,35 @@ abstract class AddonPopupBaseFragment : Fragment(), EngineSession.Observer, User
                 owner = this,
                 view = view,
             )
+
+            downloadsFeature.set(
```
Comment (jonalmeida) [inline fenix/app/src/main/java/org/mozilla/fenix/addons/AddonPopupBaseFragment.kt:67]: Is this truly safe? DownloadsFeature holds some kind of state (in my brief look) and how does that work when there are two instances of it (`BrowserFragment` and `AddonPopupBaseFragment`)?

**Review by jonalmeida — COMMENTED** [review]
I have the bandwidth for this any more, can you relay it to another teammate? 🙂

**Comment by Amejia481** [comment]
> I do not have the bandwidth for this any more, but I have left two questions to consider. Can you relay it to another teammate? 🙂

Sure, it's not a problem!

File: fenix/app/src/main/java/org/mozilla/fenix/addons/AddonPopupBaseFragment.kt (hunk 3 above)
Comment (Amejia481) [reply fenix/app/src/main/java/org/mozilla/fenix/addons/AddonPopupBaseFragment.kt:67]: We are instantiate`DownloadsFeature` in both `BrowserFragment` and `AddonPopupBaseFragment`  same happens with [promptsFeature](https://github.com/mozilla-mobile/firefox-android/pull/4845/files/3f98238951f3e470475863120997d5011516357f#diff-c1b68c76c9740c7e30d694d89f40819528bb55134ae00706bbe825b22efcb52cR39).

File: fenix/app/src/test/java/org/mozilla/fenix/browser/DownloadDialogUtilsTest.kt (hunk 2 above)
Comment (Amejia481) [reply fenix/app/src/test/java/org/mozilla/fenix/browser/DownloadDialogUtilsTest.kt]: Thanks for bringing this up I removed the spy.

**Review by rocketsroger — APPROVED** [review]
Tested and works great.  🚢

**Pull request merged**